app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuth.ts

```diff
@@ -0,0 +1,120 @@
+import { action, makeAutoObservable, observable } from 'mobx';
+import { MfaGateway } from './MfaProps';
+import { TwoFactorActivation } from './TwoFactorActivation';
+
+type TwoFactorStatus =
+  | 'two-factor-enabled'
+  | TwoFactorActivation
+  | 'two-factor-disabled';
+
+export const is2FADisabled = (s: TwoFactorStatus): s is 'two-factor-disabled' =>
+  s === 'two-factor-disabled';
+
+export const is2FAActivation = (s: TwoFactorStatus): s is TwoFactorActivation =>
+  (s as any).type === 'two-factor-activation';
+
+export const is2FAEnabled = (s: TwoFactorStatus): s is 'two-factor-enabled' =>
+  s === 'two-factor-enabled';
+
+export class TwoFactorAuth {
+  private _status: TwoFactorStatus | 'fetching' = 'fetching';
```

Comment:
Because `fetching` is not a status that the two factor authentication can be in. If you see the getter below, you'll see that `fetching` is reduced to a disabled status. If we add a spinner for an in-flight request we'd probably expose another getter as well.
Functionally it's the same, but this has a more descriptive purpose - to say that "fetching is not a TwoFactorStatus".